starter-code/src/clue.js

```diff
@@ -1,94 +1,84 @@
 
 // Weapons
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var rope = {weapon: 'rope', weight: 10};
+var knife = {weapon: 'knife', weight: 8};
+var candlestick = {weapon: 'candlestick', weight: 2};
+var dumbbell = {weapon: 'dumbbell', weight: 30};
+var poison = {weapon: 'poison', weight: 2};
+var axe = {weapon: 'axe', weight: 15};
+var bat = {weapon: 'bat', weight: 13};
+var trophy = {weapon: 'trophy', weight: 25};
+var pistol = {weapon: 'pistol', weight: 20};
+
+
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+
+var diningRoom = {room: 'Dining Room'};
+var conservatory = {room:'Conservatory'};
+var kitchen = {room: 'Kitchen'};
+var study = {room: 'Study'};
+var library = {room: 'Library'};
+var billiardRoom = {room: 'Billiard Room'};
+var lounge = {room: 'Lounge'};
+var ballroom = {room: 'Ballroom'};
+var hall = {room: 'Hall'};
+var spa = {room: 'Spa'};
+var livingRoom = {room: 'Living Room'};
+var observatory = {room: 'Observatory'};
+var theater = {room: 'Theater'};
+var guestHouse = {room: 'Guest House'};
+var patio = {room:'Patio'};
+
 
 // Characters Collection
 var charactersArray = [];
 
+charactersArray.push(mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard);
+
 // Rooms' Collection
 var roomsArray = [];
 
+roomsArray.push(diningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio);
+
 // Weapons Collection
 var weaponsArray = [];
+
+weaponsArray.push(rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol);
+
+// 
+
+var misteryEnvelope;
+
+function randomSelector(arrayOfCards){
+   if (arrayOfCards.length === 0){
+    return;
+  }else{
+  return arrayOfCards[Math.floor(Math.random() * arrayOfCards.length)];
+  }
+}
+
+function pickMistery(){
+  misteryEnvelope = [];
+  misteryEnvelope.push(randomSelector(charactersArray), randomSelector(weaponsArray), randomSelector(roomsArray));
+  return misteryEnvelope
+}
+
+
+function revealMistery (misteryEnvelope){
+
+  //console.log(misteryEnvelope);
+  result = misteryEnvelope[0].firstName + " " + misteryEnvelope[0].lastName + ' killed Mr.Boddy using the ' + misteryEnvelope[1].weapon + " in the " + misteryEnvelope[2].room;
+  return result;
+}
```

Comment:
Your getting an error here on Jasmine but when you put the code on repl.it it works fine. overall you did pass the jasmine test.